src/core/ext/xds/xds_client.cc

```diff
@@ -1383,6 +1398,8 @@ void XdsClient::ChannelState::LrsCallState::OnRequestSent(bool /*ok*/) {
 void XdsClient::ChannelState::LrsCallState::OnRecvMessage(
     absl::string_view payload) {
   MutexLock lock(&xds_client()->mu_);
+  auto cleanup =
+      absl::MakeCleanup([call = call_.get()]() { call->StartRecvMessage(); });
```

Comment:
This should be moved down below line 1404.  If we are not the current call, then we don't need to start another read.